### PR TITLE
major refactor - see README

### PR DESCRIPTION
**Pull request by Kriechi**
refactored tasks
added support for auto detecting configuration
added documentation for `rvm:hook` task
added documentation for `rvm:check` task

### CODE REVIEW

**Comment by Kriechi** [comment]
I've added a small "include guard", to only create the rvm wrapper commands once per `cap` run.

**Comment by leehambley** [comment]
This looks good to me. (//cc @kirs )

File: lib/capistrano/tasks/rvm.rake [hunk 1]
```diff
@@ -13,36 +16,49 @@ SSHKit.config.command_map = Hash.new do |hash, key|
...
 end
 
 namespace :rvm do
-  task :check_config do
+  desc "Runs the RVM hook - use it before any custom tasks if necessary"
+  task :hook do
+    unless fetch(:rvm_hooked)
+      invoke :'rvm:init'
+      invoke :'rvm:create_wrappers'
+      set :rvm_hooked, true
```
Comment (kirs) [inline lib/capistrano/tasks/rvm.rake:28]: :+1:

**Comment by kirs** [comment]
Thanks for contribution!

**Pull request merged**